pkg/chains/formats/slsa/v2alpha2/internal/resolved_dependencies/resolved_dependencies.go

```diff
@@ -0,0 +1,361 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resolvedDependencies
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/common"
+	v1 "github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/v1"
+	"github.com/tektoncd/chains/internal/backport"
+	"github.com/tektoncd/chains/pkg/artifacts"
+	"github.com/tektoncd/chains/pkg/chains/formats/slsa/attest"
+	"github.com/tektoncd/chains/pkg/chains/formats/slsa/internal/material"
+	"github.com/tektoncd/chains/pkg/chains/objects"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"go.uber.org/zap"
+	"knative.dev/pkg/logging"
+)
+
+const (
+	pipelineConfigName     = "config/pipeline"
+	taskConfigName         = "config/task"
+	pipelineTaskConfigName = "config/pipelineTask"
+	containerImageName     = "image"
+	inputResultName        = "inputs/result"
+	pipelineResourceName   = "pipelineResource"
+)
+
+// TaskRunResolvedDependencies constructs `predicate.resolvedDependencies` section by collecting all the artifacts that influence a taskrun such as source code repo and step&sidecar base images.
+func TaskRunResolvedDependencies(ctx context.Context, tro *objects.TaskRunObject) (*[]v1.ResourceDescriptor, error) {
+	var resolvedDependencies []v1.ResourceDescriptor
+
+	// add top level task config
+	if p := tro.Status.Provenance; p != nil && p.RefSource != nil {
+		rd := v1.ResourceDescriptor{
+			Name:   taskConfigName,
+			URI:    p.RefSource.URI,
+			Digest: p.RefSource.Digest,
+		}
+		resolvedDependencies = append(resolvedDependencies, rd)
+	}
+
+	// add step images
+	if err := addStepImagesToResolvedDependencies(tro.Status.Steps, &resolvedDependencies); err != nil {
+		return &resolvedDependencies, err
+	}
+
+	// add sidecar images
+	if err := addSidecarImagesToResolvedDependencies(tro.Status.Sidecars, &resolvedDependencies); err != nil {
+		return &resolvedDependencies, err
+	}
+
+	// Store git rev as ResolvedDependencies
+	gitCommit, gitURL := gitInfo(tro)
+	if gitCommit != "" && gitURL != "" {
+		resolvedDependencies = append(resolvedDependencies, v1.ResourceDescriptor{
+			Name:   inputResultName,
+			URI:    gitURL,
+			Digest: map[string]string{"sha1": gitCommit},
+		})
+		// return &resolvedDependencies, nil
+	}
+
+	// add input artifacts
+	sms := artifacts.RetrieveResolvedDependenciesFromStructuredResults(ctx, tro, artifacts.ArtifactsInputsResultName)
+	for i := range sms {
+		sms[i].Name = inputResultName
+	}
+	resolvedDependencies = append(resolvedDependencies, sms...)
+
+	// add task resources
+	addTaskResourcesToResolvedDependencies(ctx, tro, &resolvedDependencies)
+
+	// remove duplicate resolved dependencies
+	resolvedDependencies, err := removeDuplicateResolvedDependencies(resolvedDependencies)
+	if err != nil {
+		return &resolvedDependencies, err
+	}
```

Comment:
Passing the pointer to the slice to the remove function might be more efficient.